Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -25,46 +26,37 @@ class FFTModel: ObservableObject {
         }
     }
 
-    func updateAmplitudes(_ fftFloats: [Float]) {
-        var fftData = fftFloats
-        for index in 0 ..< fftData.count {
-            if fftData[index].isNaN { fftData[index] = 0.0 }
-        }
+    func updateAmplitudes(_ fftData: [Float]) {
+        var tempAmplitudes = Array(repeating: 0.0 as Float, count: numberOfBars)
+        let binsPerBar = Int((Float(fftData.count) / Float(numberOfBars)).rounded(.up))
```

Comment:
Here we figured out how many frequency bins must fit into the number of bars we are displaying so we can average them together.